test/integration/smoke/test_global_acls.py

```diff
@@ -0,0 +1,212 @@
+from marvin.lib.common import (get_domain, get_zone)
+from nose.plugins.attrib import attr
+from marvin.cloudstackException import CloudstackAPIException
+
+
+class Services:
+    """Test Global ACLs
+    """
+
+    def __init__(self):
+        self.services = {
+            "root_domain": {
+                "name": "ROOT",
+            },
+            "domain": {
+                "name": "Domain",
+            },
+            "user": {
+                "username": "user",
+                "roletype": 0,
+            },
+            "domain_admin": {
+                "username": "Domain admin",
+                "roletype": 2,
+            },
+            "root_admin": {
+                "username": "Root admin",
+                "roletype": 1,
+            },
+            "vpc": {
+                "name": "vpc-networkacl",
+                "displaytext": "vpc-networkacl",
+                "cidr": "10.1.1.0/24",
+            },
+            "vpcnetwork": {
+                "name": "vpcnetwork",
+                "displaytext": "vpcnetwork",
+            },
+            "rule": {
+                "protocol": "all",
+                "traffictype": "ingress",
+            }
+        }
+
+
+class TestGlobalACLs(cloudstackTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.testClient = super(TestGlobalACLs, cls).getClsTestClient()
+        cls.api_client = cls.testClient.getApiClient()
+
+        cls.services = Services().services
+        cls.domain = get_domain(cls.api_client)
+        cls.zone = get_zone(cls.api_client, cls.testClient.getZoneForTests())
+        return
+
+    @classmethod
+    def tearDownClass(cls):
+        super(TestGlobalACLs, cls).tearDownClass()
+
+    def setUp(self):
+        self.apiclient = self.testClient.getApiClient()
+        self.user_apiclient = self.testClient.getUserApiClient(self.services["user"]["username"],
+                                                               self.services["domain"]["name"],
+                                                               self.services["user"]["roletype"])
+
+        self.domain_admin_apiclient = self.testClient.getUserApiClient(self.services["domain_admin"]["username"],
+                                                                       self.services["domain"]["name"],
+                                                                       self.services["domain_admin"]["roletype"])
+
+        self.admin_apiclient = self.testClient.getUserApiClient(self.services["root_admin"]["username"],
+                                                                self.services["root_domain"]["name"],
+                                                                self.services["root_admin"]["roletype"])
+
+        self.cleanup = []
+        return
+
+    def tearDown(self):
+        try:
+            cleanup_resources(self.apiclient, self.cleanup)
+            cleanup_resources(self.user_apiclient, self.cleanup)
```

Comment:
I don´t think this will work. At least the list will have to be reversed to make sure no dependend resources are deleted after their dependencies which will lead to exceptions (being ignored in this case)
```suggestion
            cleanup_resources(self.apiclient, reversed(self.cleanup))
            cleanup_resources(self.user_apiclient, reversed(self.cleanup))
```